### PR TITLE
[JIT] Document torch.jit.interface

### PR DESCRIPTION
**Pull request by davidberard98**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109356

Good option for replacing "Callable" types; we should document it so
it's searchable.

### CODE REVIEW

**Comment by davidberard98** [comment]
reminder to myself to check https://docs-preview.pytorch.org/pytorch/pytorch/109356/generated/torch.jit.interface.html once the build finishes

**Review by eellison — APPROVED** [review]
nice!

File: torch/jit/_script.py [hunk 1]
```diff
@@ -1468,6 +1468,52 @@ def _check_directly_compile_overloaded(obj):
 
 
 def interface(obj):
+    r"""
+    This decorator can be used to define an interface that can be used to annotate
+    classes or modules of different types. This can be used for to annotate a submodule
+    or attribute class that could have different values, or which could be swapped
```
Comment (gmagogsfm) [inline torch/jit/_script.py:1474]:
```suggestion
    or attribute class that could have different types that implement the same interface, or which could be swapped
```

**Review by gmagogsfm — APPROVED** [review]

**Pull request closed without merging**